app/views/training_status/show.json.jbuilder

```diff
@@ -2,12 +2,14 @@
 
 json.course do
   json.training_modules @assigned_training_modules do |training_module|
-    training_progress_manager = TrainingProgressManager.new(@user, training_module)
+    tp_manager = TrainingProgressManager.new(@user, training_module)
 
     json.id training_module.id
     json.module_name training_module.name
     json.kind training_module.kind
-    json.status training_progress_manager.status
-    json.completion_date training_progress_manager.completion_date
+    json.status tp_manager.status
+    json.completion_date tp_manager.completion_date
+    tmu = TrainingModulesUsers.find_by(user_id: @user.id, training_module_id: training_module.id)
```

Comment:
This is probably an extra query that doesn't need to happen; the `TrainingProgressManager` already finds the `TrainingModulesUsers` when it is initialized. You could add a `completion_time` method to `TrainingProgressManager`.

Reply:
Okay!